apps/web/src/features/walletconnect/WalletConnectContext.tsx

```diff
@@ -1,23 +1,344 @@
+
+        // Send response to WalletConnect
+        await walletConnect.sendSessionResponse(topic, response)
+      } catch (e) {
+        setError(e as Error)
+      }
+    })
+  }, [walletConnect, chainId, safeWalletProvider])
+
+  //
+  // --- One-click Auth
+  //
+  useEffect(() => {
+    if (!walletConnect || !safeWalletProvider || !chainId) return
+
+    return walletConnect.onSessionAuth(async (event) => {
+      const { authPayload, requester } = event.params
+      const peerName = getPeerName(requester) || FALLBACK_PEER_NAME
+
+      if (!IS_PRODUCTION) {
+        console.log('[WalletConnect] auth', authPayload, requester)
+      }
+
+      if (!authPayload.chains.includes(getEip155ChainId(chainId))) {
+        setError(getWrongChainError(peerName))
+        return
+      }
+
+      const getSignature = async () => {
+        const message = walletConnect.formatAuthMessage(authPayload, chainId, safeAddress)
+
+        if (!IS_PRODUCTION) {
+          console.log('[WalletConnect] SiWE message', message)
+        }
+
+        const appInfo = {
+          url: LEGACY_WC_APP_URL, // required for server-side analytics
+          name: peerName,
+          description: requester.metadata.description,
+          iconUrl: requester.metadata.icons[0],
+        }
+
+        return safeWalletProvider.request(
+          event.id,
+          {
+            method: 'personal_sign',
+            params: [message, safeAddress],
+          },
+          appInfo,
+        )
+      }
+
+      // Close the popup
+      setIsLoading(WCLoadingState.APPROVE)
+      setOpen(false)
+
+      // Get a signature and send it to WalletConnect
+      try {
+        const signature = await getSignature()
+        if ('error' in signature) throw new Error(signature.error.message)
+        await walletConnect.approveSessionAuth(event.id, authPayload, signature.result as string, chainId, safeAddress)
+      } catch (e) {
+        try {
+          await walletConnect.rejectSessionAuth(event.id)
+        } catch (err) {
+          e = err
+        }
+        setError(e as Error)
+        setOpen(true)
+      }
+
+      setIsLoading(undefined)
+    })
+  }, [walletConnect, safeWalletProvider, chainId, safeAddress, setOpen])
+
+  //
+  // --- Sessions
+  //
+  const [sessions, setSessions] = useState<SessionTypes.Struct[]>([])
+
+  const updateSessions = useCallback(() => {
+    walletConnect && setSessions(walletConnect.getActiveSessions())
+  }, [walletConnect])
+
+  // Initial sessions
+  useEffect(updateSessions, [updateSessions])
+
+  // On session add
+  useEffect(() => {
+    return walletConnect?.onSessionAdd(updateSessions)
+  }, [walletConnect, updateSessions])
+
+  // On session delete
+  useEffect(() => {
+    return walletConnect?.onSessionDelete(updateSessions)
+  }, [walletConnect, updateSessions])
+
+  //
+  // --- Proposals
+  //
+  const [sessionProposal, setSessionProposal] = useState<WalletKitTypes.SessionProposal | null>(null)
```

Comment:
Is there a reason why this was changed from undefined to null?

Reply:
The context had most values as `null` by default. Only isLoading was `undefined`. I'd rather change the latter to `false` by default.